django/forms/formsets.py

```diff
@@ -470,7 +470,5 @@ def formset_factory(form, formset=BaseFormSet, extra=1, can_order=False,
 
 def all_valid(formsets):
     """Validate every formset and return True if all are valid."""
-    valid = True
-    for formset in formsets:
-        valid &= formset.is_valid()
-    return valid
+    # Not using the generator expression is deliberate.
```

Comment:
For what reason? Doesn't `all()` work fine on generators? (Typically we still use list comprehensions for `str.join()` as it converts to a list internally anyway.)

Reply:
```suggestion
    # List comprehension ensures is_valid() is called for all formsets.
```